opal/mca/btl/smcuda/btl_smcuda_accelerator.c

```diff
@@ -39,50 +40,62 @@ static bool smcuda_accelerator_initialized = false;
 
 void mca_btl_smcuda_accelerator_fini(void);
 
-int mca_btl_smcuda_accelerator_init(void)
+/* Initialize the internal ipc stream and the events (s&e) */
+static int mca_btl_smcuda_accelerator_sae_init(void)
 {
     int rc = OPAL_SUCCESS;
     int i;
-    OBJ_CONSTRUCT(&btl_smcuda_accelerator_ipc_lock, opal_mutex_t);
-    /* The first available status index is 0.  Make an empty frag
-       array. */
 
     rc = opal_accelerator.create_stream(MCA_ACCELERATOR_NO_DEVICE_ID, &ipc_stream);
     if (OPAL_SUCCESS != rc) {
         opal_output_verbose(1, mca_btl_smcuda_component.cuda_ipc_output, "Failed to create accelerator ipc_stream stream.");
-        goto cleanup_and_error;
+        return OPAL_ERROR;
+    }
+
+    /* Create the events since they can be reused. */
+    for (i = 0; i < accelerator_event_max; i++) {
+        rc = opal_accelerator.create_event(MCA_ACCELERATOR_NO_DEVICE_ID, &accelerator_event_ipc_array[i], opal_accelerator_use_sync_memops ? false : true);
```

Comment:
I'm slightly confused here about what this PR really does. You mentioned it improves the handling of the process GPU/stream, but I see here that the events are created without a device association (MCA_ACCELERATOR_NO_DEVICE_ID) as if they were expected to be generic (which is not the case, at least not with CUDA)?

Reply:
Moreover, the CUDA accelerator simply ignores the device id, which either suggest that we don't need it or that the CUDA accelerator framework is careless on the handling of the devices.

Reply:
that would be awesome, and will make the code much more readable.